packages/api/src/bsky-agent.ts

```diff
@@ -406,6 +436,46 @@ export class BskyAgent extends AtpAgent {
         .concat([personalDetailsPref])
     })
   }
+
+  async setFeedViewPrefs(
+    feed: string,
+    pref: Omit<AppBskyActorDefs.FeedViewPref, '$type' | 'feed'>,
+  ) {
+    await updatePreferences(this, (prefs: AppBskyActorDefs.Preferences) => {
+      const existing = prefs.findLast(
+        (pref) =>
+          AppBskyActorDefs.isFeedViewPref(pref) &&
+          AppBskyActorDefs.validateFeedViewPref(pref).success &&
+          pref.feed === feed,
+      )
+      if (existing) {
+        pref = { ...existing, ...pref }
```

Comment:
I'm not super strong with typescript, or our use of schema objects, but i'd worry that this might not successfully remove/nullify preferences in some situations.

Eg, if you had `hideRepliesByLikeCount = 10`, then submitted a request where `hideRepliesByLikeCount` was cleared (but not specifically set to `0`), would the old value be retained?

Reply:
Good looking out but it actually does exactly what's intended

```
setFeedViewPrefs({hideRepliesByLikeCount: 5}) -> 5
setFeedViewPrefs({hideRepliesByLikeCount: undefined}) -> undefined
setFeedViewPrefs({}) -> no change
```